priam/src/main/java/com/netflix/priam/resources/BackupServletV2.java

```diff
@@ -181,4 +193,26 @@ public Response list(@PathParam("daterange") String daterange) throws Exception
                                                 .collect(Collectors.toList())))
                 .build();
     }
+
+    @GET
+    @Path("/state/{hours}")
+    public Response backupState(@PathParam("hours") int hours) throws Exception {
+        Map<String, Object> responseMap = new HashMap<>();
+
+        responseMap.put("tasksQueued", fs.getUploadTasksQueued());
+        responseMap.put("queueSize", priamServer.getConfiguration().getBackupQueueSize());
```

Comment:
This does not require information that only the local node has. Please read it from fast properties. Or better yet, don't include it in the calculations as we might move to a model where the queue is small and populated as needed and the only metric that matters is the remaining files on disk.

Reply:
This FP is not set for every cluster and we need it to compute the file threshold for which we want to wait for backup upload to complete.